ww4ff/data/dataset/dataset.py

```diff
@@ -138,9 +138,24 @@ def _inc(self):
             self._init_frame_labels()
             return self.dataset.dataset[self.curr_file_idx]
 
+        def emit_next_positive_example(self):
+            example = self.dataset.dataset[self.curr_file_idx]
+            pos_ts, pos_lbl = self.frame_labels[self.label_pointer]
+            self.output_positives.append(pos_ts)
+            b = int(pos_ts / 1000 * self.dataset.dataset.sr)
+            a = max(b - self.dataset.window_size, 0)
+            self.label_pointer += 1
+            return ClassificationBatch.from_single(example.audio_data[..., a:b], pos_lbl)
+
         def __next__(self) -> ClassificationBatch:
```

Comment:
Can you explain this next code little?
it's hard to follow and understand

Reply:
Yeah, I should document everything at some point. For evaluation, we stride a window (from left to right) across the entire dataset, concatenating files as we go and outputting an example with the appropriate label for each window.